src/python/pants/option/option_types.py

```diff
@@ -660,19 +664,52 @@ def _convert_(self, val: Any) -> dict[str, _ValueT]:
 # -----------------------------------------------------------------------------------------------
 # "Specialized" Concrete Option Classes
 # -----------------------------------------------------------------------------------------------
+def _get_tool_name(subsystem_cls: type["Subsystem"]):
+    return getattr(subsystem_cls, "name", subsystem_cls.options_scope)
```

Comment:
Thoughts on not falling back to the options scope? I'm thinking that if you don't have `name` specified, then maybe we eagerly error if you have not passed in the kwarg `tool_name`. 

My fear is plugin authors not realizing they're supposed to be doing this so they end up with a subpar message. 

--

I originally was worried about Name vs options_scope, but I think I support it here. Basically, I think where I fall down is that we should always use options_scope when the user is going to refer to thing, e.g. `--lint-only` `./pants help tools` keys. But otherwise, name is great.

I'm not worried about the one extra line required by plugin authors. What I care about is end users having a nice experience.

Reply:
That makes sense. We also have the option of calling this `visible_name` or something along those lines to make it more clear what the string is used for.

Reply:
(or `display_name`.)

One thing I don't like about this change is only some subsystems have `name`, while others don't. It feels inconsistent (although ack that, again, the name already existed hardcoded, just in a string).

Going to use `display_name` for now.